tests/unit/utils/test_objectives.py

```diff
@@ -72,6 +76,39 @@ def test_logarithmic_goldstein_price_no_function_values_are_less_than_global_min
     )
 
 
+@pytest.mark.parametrize(
+    "obj_inst, input_dim, num_obj, gen_pf_num",
+    [
+        (DTLZ1(3, 2), 3, 2, 1000),
+        (DTLZ1(5, 3), 5, 3, 1000),
+        (DTLZ2(3, 2), 3, 2, 1000),
+        (DTLZ2(12, 6), 12, 6, 1000),
+    ],
+)
+def test_gen_pareto_front_is_equal_to_math_defined(
+    obj_inst: MultiObjectiveTestProblem, input_dim: int, num_obj: int, gen_pf_num: int
+):
+    pfs = obj_inst.gen_pareto_optimal_points(gen_pf_num)
+    if isinstance(obj_inst, DTLZ1):
+        tf.assert_equal(tf.reduce_sum(pfs, axis=1), 0.5)
+    elif isinstance(obj_inst, DTLZ2):
+        tf.debugging.assert_near(tf.norm(pfs, axis=1), 1.0, rtol=1e-6)
+
+
+@pytest.mark.parametrize(
+    "obj_inst, actual_x",
+    [
+        (DTLZ1(3, 2), tf.constant([[0.3, 0.1]])),
+        (DTLZ2(5, 2), tf.constant([[0.3, 0.1]])),
+    ],
+)
+def test_func_raises_specified_input_dim_not_alin_with_actual_input_dim(
+    obj_inst: MultiObjectiveTestProblem, actual_x: TensorType
+):
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        obj_inst.prepare_benchmark()(actual_x)
+
+
```

Comment:
Can we also test a specific input and see that it gives the expected outputs? i.e. similar to the tests for the non-MO synthetic functions.

Reply:
Some numerical tests added. 
Nevertheless, the expected results in the test are hand derived, a bit unintuitive for maintaining I guess.